### PR TITLE
[sending] Add support for 'sending'

### PR DESCRIPTION
**Pull request by gottesmm**
This PR adds support to the compiler for 'sending'. It does not completely remove 'transferring' (since it may already be used). Instead what it does is:

1. Internally except for in the parser and the clang importer, we only represent
'sending'. This means that it will be easy to remove 'transferring' once enough
time has passed.

2. I included a warning that suggested to the user to change 'transferring' ->
'sending'.

3. I duplicated the parsing diagnostics for 'sending' so both will still get
different sets of diagnostics for parsing issues... but anywhere below parsing,
I have just changed 'transferring' to 'sending' since transferring isn't
represented at those lower levels.

4. Since SendingArgsAndResults is always enabled when TransferringArgsAndResults
is enabled (NOTE not vis-a-versa), we know that we can always parse sending. So
we import "transferring" as "sending". This means that even if one marks a
function with "transferring", the compiler will guard it behind a
SendingArgsAndResults -D flag and in the imported header print out sending.

rdar://128216574

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

File: include/swift/Basic/Features.def [hunk 1]
```diff
@@ -353,7 +353,14 @@ EXPERIMENTAL_FEATURE(FixedArrays, true)
...
 // Allow for the 'transferring' keyword to be applied to arguments and results.
-SUPPRESSIBLE_EXPERIMENTAL_FEATURE(TransferringArgsAndResults, true)
+//
+// Enables SendingArgsAndResults as well. After parsing, we just represent this
+// as 'sendable' implying that since both are always enabled together, this
+// doesn't need to be suppressed.
+EXPERIMENTAL_FEATURE(TransferringArgsAndResults, true)
+
+// Allow for the 'sending' keyword to be applied to arguments and results.
+SUPPRESSIBLE_EXPERIMENTAL_FEATURE(SendingArgsAndResults, true)
```
Comment (hborla) [inline include/swift/Basic/Features.def:363]: These should not be experimental features

**Review by ktoso — APPROVED** [review]
LGTM, when do you think we'll remove `transfering` - will we be able to drop it in the final 6.0 or may have to tag it along longer? 🤔 

Cool thx, when this lands I'll follow up with some changes in distributed

**Comment by ahoppen** [comment]
I think this also needs a corresponding parser change in swift-syntax.

**Comment by gottesmm** [comment]
apple/swift-syntax#2658

@swift-ci smoke test

**Comment by gottesmm** [comment]
https://github.com/apple/swift-syntax/pull/2658

@swift-ci smoke test

**Pull request merged**